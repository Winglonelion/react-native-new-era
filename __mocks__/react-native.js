import { Animated as RNAnimated } from 'react-native';

export const Animated = {
  ...RNAnimated,
  parallel: () => ({
    // immediately invoke callback
    start: cb => cb(),
  }),
  timing: () => ({
    // immediately invoke callback
    start: cb => cb(),
  }),
};
