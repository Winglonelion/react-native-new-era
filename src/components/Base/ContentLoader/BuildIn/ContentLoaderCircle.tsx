import React, { memo } from 'react';
import ContentLoader, { Circle } from '../index';
import { IContentLoaderProps } from '../ContentLoader.types';
import { LOADING_SPEED } from './config';

export interface PropTypes extends IContentLoaderProps {
  ready?: boolean;
  children: any;
  offsetX?: number;
  offsetY?: number;
  size: number;
}

const ContentLoaderCircle: React.FC<PropTypes> = memo(props => {
  const {
    children,
    ready = false,
    size = 100,
    offsetX,
    offsetY,
    ...rest
  } = props;

  if (ready) return children;

  return (
    <ContentLoader
      speed={LOADING_SPEED}
      width={props.width ?? 100}
      height={props.height ?? 100}
      viewBox={`0 0 ${props.width || 100} ${props.height || 100}`}
      backgroundColor="#f3f3f3"
      foregroundColor="#ecebeb"
      {...rest}>
      <Circle
        cx={offsetX ?? size / 2}
        cy={offsetY ?? size / 2}
        r={(size ?? props.width ?? 100) / 2}
      />
    </ContentLoader>
  );
});

export default ContentLoaderCircle;
