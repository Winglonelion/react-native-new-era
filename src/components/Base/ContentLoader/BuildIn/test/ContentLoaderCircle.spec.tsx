import React from 'react';
import { shallow } from 'enzyme';
import ContentLoaderCircle, { PropTypes } from '../ContentLoaderCircle';
import { View } from 'react-native';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<ContentLoaderCircle {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('ContentLoaderCircle check snapshot', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup({
      children: <View />,
      size: 30,
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('ContentLoaderCircle should match snapshot with offsets', () => {
    const { wrapper } = setup({
      children: <View />,
      offsetX: 100,
      offsetY: 100,
      size: 30,
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('ContentLoaderCircle should match snapshot with ready', () => {
    const { wrapper } = setup({
      children: <View />,
      ready: true,
      size: 30,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
