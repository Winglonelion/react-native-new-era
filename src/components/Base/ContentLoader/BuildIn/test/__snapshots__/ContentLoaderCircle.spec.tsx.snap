// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentLoaderCircle check snapshot ContentLoaderCircle should match snapshot with offsets 1`] = `
<ContentLoader
  backgroundColor="#f3f3f3"
  foregroundColor="#ecebeb"
  height={100}
  speed={1.5}
  viewBox="0 0 100 100"
  width={100}
>
  <Circle
    cx={100}
    cy={100}
    r={15}
  />
</ContentLoader>
`;

exports[`ContentLoaderCircle check snapshot ContentLoaderCircle should match snapshot with ready 1`] = `<View />`;

exports[`ContentLoaderCircle check snapshot should match snapshot 1`] = `
<ContentLoader
  backgroundColor="#f3f3f3"
  foregroundColor="#ecebeb"
  height={100}
  speed={1.5}
  viewBox="0 0 100 100"
  width={100}
>
  <Circle
    cx={15}
    cy={15}
    r={15}
  />
</ContentLoader>
`;
